edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/CloudEndpoint.cs

```diff
@@ -83,8 +88,13 @@ public CloudMessageProcessor(CloudEndpoint endpoint)
                     {
                         try
                         {
-                            await cp.SendMessageAsync(message);
+                            string id = this.GetIdentity(routingMessage).Expect(() => new InvalidOperationException("Could not retrieve identity of message"));
+                            using (Metrics.CloudLatency(this.cloudEndpoint.metricsCollector, id))
```

Comment:
Will this add overhead if metricsCollector is None?

Reply:
Some, but shouldn't be too bad.